### PR TITLE
Enable CodeQL GitHub code scanning

### PR DESCRIPTION
**Pull request by raphsutti**

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+name: 'CodeQL'
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: '0 13 * * 6'
```
Comment (72636c) [inline .github/workflows/codeql-analysis.yml:12]: Looks like the schedule is randomised across the repos at the moment. Do you think that makes sense, or would it be better for us to get all the notifications at once?

**Review by 72636c — APPROVED** [review]

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (raphsutti) [reply .github/workflows/codeql-analysis.yml:12]: A specific scan day/time make sense 🤔  
What day makes sense? Something like Every Monday at 9am?

Comment (72636c) [reply .github/workflows/codeql-analysis.yml:12]: Sounds good 👍

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ on:
   push:
   pull_request:
   schedule:
-    - cron: '0 13 * * 6'
+    - cron: '0 9 * * 1'
```
Comment (72636c) [inline .github/workflows/codeql-analysis.yml:12]: Is this in UTC? 😛

**Review by 72636c — APPROVED** [review]

File: .github/workflows/codeql-analysis.yml (hunk 2 above)
Comment (raphsutti) [reply .github/workflows/codeql-analysis.yml:12]: Timezones 🤦  yes it is 
https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows

**Review by 72636c — APPROVED** [review]
🚀 thanks for this

**Pull request merged**